### PR TITLE
[nnfw] Set accelerator for nnfw

### PR DESCRIPTION
**Pull request by kparichay**
Allow setting of accelerator for nnfw
Parsing accelerator moved to tensor_filter_common. Currently, used in
tflite and nnfw, and use directly for other filters like this.

Related Issue: #1849 

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite_core.cc [hunk 1]
```diff
@@ -431,40 +431,17 @@ TFLiteCore::TFLiteCore (const char * _model_path, const char * accelerators)
...
-        g_free (word);
-        break;
-      }
-    }
-    g_match_info_free (match_info);
-    g_regex_unref (nnapi_elem);
-  } else  {
-    goto use_nnapi_ini;
-  }
+  else
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite_core.cc]: It seems `else` is not necessary.

Comment (kparichay) [reply ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite_core.cc]: updated to set `use_nnapi = TRUE` at the start of the function.

**Comment by myungjoo** [comment]
Could you please add the following diff to PROP_ACCELEARTTOR:
```
diff --git a/gst/nnstreamer/tensor_filter/tensor_filter_common.c b/gst/nnstreamer/tensor_filter/tensor_filter_common.c
index 8a6ac45..7edf4b8 100644
--- a/gst/nnstreamer/tensor_filter/tensor_filter_common.c
+++ b/gst/nnstreamer/tensor_filter/tensor_filter_common.c
@@ -268,7 +268,8 @@ gst_tensor_filter_install_properties (GObjectClass * gobject_class)
           "(true/false):(comma separated ACCELERATOR(s)). "
           "true/false determines if accelerator is to be used. "
           "list of accelerators determines the backend (ignored with false). "
-          "Example, if GPU, NPU can be used but not CPU - true:(GPU,NPU,!CPU).",
+          "Example, if GPU, NPU can be used but not CPU - true:(GPU,NPU,!CPU). "
+          "Note that only a few subplugins support this property.",
           "", G_PARAM_READWRITE));
   g_object_class_install_property (gobject_class, PROP_IS_UPDATABLE,
       g_param_spec_boolean ("is-updatable", "Updatable model",
```

**Comment by kparichay** [comment]
V2:
Added note that PROP_ACCELERATOR is supported in a few filter extensions only.

**Comment by kparichay** [comment]
> Please add corresponding test cases (including Tizen CAPI) in later PRs.

Yes, will do

**Pull request merged**